src/Framework/AspNetCoreAnalyzers/src/Analyzers/DiagnosticDescriptors.cs

```diff
@@ -79,4 +79,14 @@ internal static class DiagnosticDescriptors
         DiagnosticSeverity.Error,
         isEnabledByDefault: true,
         helpLinkUri: "https://aka.ms/aspnet/analyzers");
+
+    internal static readonly DiagnosticDescriptor DisallowConfigureAppConfigureHostBuilder = new(
+        "ASP0013",
+        "Do not use builder.Host.ConfigureAppConfiguration or builder.Host.ConfigureHostConfiguration",
+        "Replace with builder.Configuration",
+        "Usage",
+        DiagnosticSeverity.Error,
```

Comment:
I'd make this one a warning like #42354 and #42356 as this doesn't functionally break an application if it's used.